### PR TITLE
rcache/udreg: make reference count thread safe

### PR DESCRIPTION
**Pull request by hjelmn**
Back-ported from master.

Signed-off-by: Nathan Hjelm hjelmn@lanl.gov

(cherry picked from commit open-mpi/ompi@4658b761e4dd86d64bca0c9ed9b153dff8846903)

Signed-off-by: Nathan Hjelm hjelmn@lanl.gov

### CODE REVIEW

**Comment by hjelmn** [comment]
:bot:assign: @hppritcha 
:bot:milestone:v2.0.1
:bot:label:bug

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1991/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1993/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1992/ for details.

File: opal/mca/mpool/udreg/mpool_udreg_module.c [hunk 1]
```diff
@@ -490,10 +493,6 @@ int mca_mpool_udreg_deregister(struct mca_mpool_base_module_t *mpool,
 {
     mca_mpool_udreg_module_t *mpool_udreg = (mca_mpool_udreg_module_t *) mpool;
```
Comment (hppritcha) [inline opal/mca/mpool/udreg/mpool_udreg_module.c:495]: where is the ref_count for the ref object now being decremented?

Comment (hjelmn) [inline opal/mca/mpool/udreg/mpool_udreg_module.c:495]: In mca_mpool_udreg_dereg_func above.

**Comment by hppritcha** [comment]
I see thanks.
👍

**Comment by hppritcha** [comment]
@jsquyres this is ready to merge.

**Pull request merged**